File: 0383-ransom-note/0383-ransom-note.cpp
class Solution {
public:
    bool canConstruct(string ransomNote, string magazine) {
        int rSize = ransomNote.size(), mSize = magazine.size();
        if(rSize > mSize) return false;
        sort(ransomNote.begin(), ransomNote.end());
        sort(magazine.begin(), magazine.end());
        int index = 0;
        for(int i=0;i<mSize;i++){
            if(ransomNote[index] == magazine[i]){
                index++;
            }
            if(index == rSize) break;;
        }

        return index == rSize;
    }
};

File: 0383-ransom-note/0383-ransom-note.java
class Solution {
    public boolean canConstruct(String ransomNote, String magazine) {
        int rSize = ransomNote.length();
        int mSize = magazine.length();
        if(rSize > mSize) return false;
        int index = 0;
        char[] rArr = ransomNote.toCharArray();
        Arrays.sort(rArr);
        char[] mArr = magazine.toCharArray();
        Arrays.sort(mArr);
        for(int i=0;i<mSize;i++){
            if(index == rSize) break; 
            else if(rArr[index] == mArr[i]){
                index++;
            }
        }
        return index == rSize;
    }
}